overlord/devicestate/devicestate.go

```diff
@@ -185,8 +185,8 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, flags sn
 		return fmt.Errorf("cannot install %s snap on classic if not requested by the model", kind)
 	}
 
-	if snapInfo.Name() != expectedName {
-		return fmt.Errorf("cannot install %s %q, model assertion requests %q", kind, snapInfo.Name(), expectedName)
+	if snapInfo.InstanceName() != expectedName {
```

Comment:
CC @pedronis do we envision instance installation of non-app snaps? (gadget, kernel, base). I think we should forbid that.

Reply:
there might be a use case to install the same base from different tracks maybe. but parallel installs of kernel and gadget don't make sense